### DISCUSSION TITLE
How can remove the warnings that comes when I'm testing

### DISCUSSION THREAD

**Discussion by ArnaudBand** [Cyfrin/foundry-full-course-cu#1499] [Q&A] [answered]
Hello Devs,

I'm currently working on the _Solidity fundamental Section 2: Fund Me_. I have a question regarding removing warnings in my test output. When I run `forge test --fork-url $MAINNET_RPC_URL`, I receive the following warnings:

```bash
Warning (2018): Function state mutability can be restricted to view
  --> test/FundMeTest.t.sol:17:5:
   |
17 |     function testMinimumDollarIsFive() public {
   |     ^ (Relevant source part starts here and spans across multiple lines).

Warning (2018): Function state mutability can be restricted to view
  --> test/FundMeTest.t.sol:21:5:
   |
21 |     function testIsOwner() public {
   |     ^ (Relevant source part starts here and spans across multiple lines).

Warning (2018): Function state mutability can be restricted to view
  --> test/FundMeTest.t.sol:25:5:
   |
25 |     function testPriceFeedVersionIsAccurate() public {
   |     ^ (Relevant source part starts here and spans across multiple lines).
```

I would appreciate any guidance on how to resolve these warnings. You can find the full code in my repository: [link](https://github.com/ArnaudBand/cyfrin_updraft_course/tree/main/foundry-fund-me).

Thank you for your assistance.

**Comment by EngrPips** [comment, marked as answer]
Hello @ArnaudBand , all you have to do is add `view` to the functions that the compiler suggest you make a view functions and the warning would go away for example adding view to the functions like i demonstrated below

```javascript
 function testMinimumDollarIsFive() public view {
```

**Reply by ArnaudBand** [reply]
Thank you for your support

**Reply by EngrPips** [reply]
I am glad i could be of help

**Comment by 0x-codeX** [comment]
Hi, you can write the test functions like so, 

`function testMinimumDollarIsFive() public view {
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }`

However some other function are pure instead of view, but view should work fine with your current predicament.

I hope it helps.

**Reply by ArnaudBand** [reply]
Thank you for your help!
